### PR TITLE
Add  alfresco module structure + basic sender

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: file/file-alfresco/pom.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        <dependency>
+            <groupId>javax.enterprise</groupId>
+            <artifactId>cdi-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
```
Comment (matt-rich) [inline file/file-alfresco/pom.xml]: Can we group dependencies based on provided / project / 3rd party / testing etc so it is easier to see duplicates.

File: file/file-alfresco/pom.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.sun.xml.bind</groupId>
+                    <artifactId>jaxb-impl</artifactId>
+                </exclusion>
+
+            </exclusions>
+        </dependency>
+        <dependency>
```
Comment (matt-rich) [inline file/file-alfresco/pom.xml]: if we have this do we still need the junit dependency?

File: file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/rest/AlfrescoRestClient.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    private Client getClient() {
+        Client client;
+
+        if ("none".equals(proxyType)) {
+            client = ClientBuilder.newClient();
+        } else {
+            client = new ResteasyClientBuilder()
+                    .defaultProxy(proxyHostname, Integer.parseInt(proxyPort), proxyType)
```
Comment (matt-rich) [inline file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/rest/AlfrescoRestClient.java]: static import

File: file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.MultivaluedHashMap;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@ApplicationScoped
+public class AlfrescoFileSender implements FileSender {
+
+    private static final String CPPUID = "cppuid";
```
Comment (matt-rich) [inline file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java]: is that a constant elsewhere? - is this still the correct header name?

File: file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    ObjectMapper objectMapper;
+
+    @Inject
+    AlfrescoRestClient restClient;
+
+
+    @Override
+    public FileData send(final String fileName, final byte[] content) {
+        final MultivaluedHashMap<String, Object> headers = new MultivaluedHashMap<>();
+        headers.put(CPPUID, Collections.singletonList(alfrescoUploadUser));
```
Comment (matt-rich) [inline file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java]: static import

File: file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java (hunk 4 above)
Comment (jaceko) [inline file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java]: This is what the material context is sending so I presume it's been configured somewhere in Alfresco

**Pull request merged**